### PR TITLE
Possible implementation of a configurable ani spam filter

### PR DESCRIPTION
**Pull request by Romanow88**

### CODE REVIEW

File: antispam.py [hunk 1]
```diff
@@ -27,30 +29,39 @@ def is_spam(self, identifier):
...
-def spam_protect(func):
-    def call(*args, **kwargs):
-        if len(args) is 3:
-            update = args[2]
-            if AntiSpam().is_spam(update.message.from_user.id):
-                update.message.reply_text("You are too talkative, pls wait a bit and try again")
-                return
-        return func(*args, **kwargs)
-    return call
+        for instance in AntiSpam.__refs__:
```
Comment (alexandra-trisch) [inline antispam.py]: this looks kind of dangerous to me
` The original object can be retrieved by calling the reference object if the referent is still alive; if the referent is no longer alive, calling the reference object will cause None to be returned. `
and moreover the object can be destroyed by the GC at any time, how can we guarantee that it's still alive. Documentation says about callback for this case. But still looks UB

Comment (Romanow88) [reply antispam.py]: I see, I assumed there would be some locking involved when iterating.

As a solution:

```
        instanses = set([])
        instanses .update(_dangling)
        for instance in instances ....
```

**Comment by Romanow88** [comment]
Just as a side note, the current impl always creates a new antispam object when decorating a function. This means that the functions have separate spam counters, so that the usage of one function does not contribute the to usage of another. Is that acceptable?

**Comment by alexandra-trisch** [comment]
it looks over complicated for me. Maybe the option of going back to have static variable `_lookup` in class is not so bad after all

**Comment by Romanow88** [comment]
Alright :)

**Pull request merged**